File: Generating/factory/cpp/BitcoinMarket.cpp
#ifndef bitcoin_instance_hpp
#define bitcoin_instance_hpp

#include "CoinInstance.hpp"
#include "MarketFactory.cpp"
#include <string>

using namespace std;

class Bitcoin : public CoinInstance
{
  public:
    Bitcoin(void) {
      this->name = "Bitcoin";
      this->network = "testnet";
      this->balance = 0;
    };
    ~Bitcoin(void) {};
    string getName() {
      return this->name;
    };

    int getBalance() {
      return this->balance;
    };

    string getNetwork() {
      return this->network;
    };

    void setBalance(int new_balance) {
      this->balance = new_balance;
    };
};

class BitcoinMarket : public MarketFactory
{
  public:
    CoinInstance * getCoinInstance() {
      return new Bitcoin();
    };
};

#endif

File: Generating/factory/python/BitcoinMarket.py
from MarketFactory import MarketFactory

class Bitcoin:
  params = None

  def __init__(self, params):
    self.name = 'Bitcoin'
    self.params = params
    self.balance = 0
    pass
  
  def getName(self):
    return self.name

  def getNetwork(self):
    return self.params['configuration']['network']

  def getBalance(self):
    return self.balance

  def setBalance(self, newBalanceValue):
    self.balance = newBalanceValue
    pass

class BitcoinMarket(MarketFactory):
  params = None

  def __init__(self, params):
    self.params = params
    pass

  def getCoinInstance(self):
    return Bitcoin(self.params)
